src/components/CompareResults/ResultsTable.tsx

```diff
@@ -107,10 +107,43 @@ export default function ResultsTable() {
     new Map() as Map<string, Set<string>>, // ColumnID -> Set<Values to remove>
   );
 
+  const accessFilterValues = () => {
+    cellsConfiguration.forEach(({ key }) => {
+      const searchValue = [...(tableFilters.get(key) ?? [])];
+
+      if (searchValue.length > 0) {
+        rawSearchParams.set(`filter-${key}`, searchValue.join(','));
+      } else {
+        rawSearchParams.delete(`filter-${key}`);
+      }
+    });
+    updateRawSearchParams(rawSearchParams);
+  };
+  const accessURlParams = () => {
+    const initialFilters = new Map() as Map<string, Set<string>>;
+    cellsConfiguration.forEach(({ key, possibleValues }) => {
+      if (!possibleValues) return;
+      const paramValue = rawSearchParams.get(`filter-${key}`);
+      initialFilters.set(key, new Set(paramValue?.split(',')));
+    });
+    setTableFilters(initialFilters);
+  };
+
+  useEffect(() => {
+    accessURlParams();
+  }, []);
```

Comment:
Instead of using an effect, it would be better to use the 3rd version of useState (as described in the react documentation: https://react.dev/reference/react/useState#usestate).

This would set the initial value of `tableFilters` by using the URL Params directly.

By using an effect, instead, this impairs the performance. Indeed this works like this:
1. React renders the component (and the rest of the tree)
2. React runs the effect.
3. If the effect changes the state, React renders the component again... and the rest of the tree as well!

We want to avoid 2 and 3. By setting the right state right away at the first render time, only the first step is run, so this is much faster.

I'd like to stress that you need to use the 3rd version of `useState`, not the first or second. Why? Because we want the computation to happen just once. This is explained in this part of their documentation: https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state

Reply:
@julienw Thank you for the detailed explanation! 

I’ve updated the code to use lazy initialization for tableFilters based on the URL Params as suggested. This way, the computation only happens once, and React renders the component only once during the initial render.

Thanks again for pointing this out, and for the useful documentation links!

Reply:
Hi @julienw I have implemented these changes and waiting for a re-review